lib/graphql_schema.rb

```diff
@@ -2,6 +2,24 @@
 require 'set'
 
 class GraphQLSchema
+  INVALID_SCHEMA = Class.new(StandardError)
+
+  class << self
+    def load_schema(schema)
+      case schema
+      when Hash
+        new(schema)
+      when GraphQLSchema
+        schema
+      when Pathname, String
+        schema = File.read(schema) if Pathname(schema).file?
```

Comment:
Well URL support could easily be added regardless. In my opinion the point of this was flexibility, but I'm fine leaving it out as is.